### PR TITLE
[SHT_LLVM_BB_ADDR_MAP] Add assertion and clarify docstring

### PR DESCRIPTION
**Pull request by boomanaiden154**
This patch adds an assertion to readBBAddrMapImpl to confirm that PGOAnalyses and BBAddrMaps are of the same size when PGO information is requested (part of the API contract). This patch also updates the docstring for readBBAddrMap to better clarify what is guaranteed.

### CODE REVIEW

**Comment by boomanaiden154** [comment]
Bump on this patch when reviewers have a chance. Thanks!

File: llvm/include/llvm/Object/ELFObjectFile.h [hunk 1]
```diff
@@ -110,11 +110,12 @@ class ELFObjectFileBase : public ObjectFile {
...
-  // corresponding to the section with that index. When `PGOAnalyses`is
-  // specified, the vector is cleared then filled with extra PGO data.
-  // `PGOAnalyses` will always be the same length as the return value on
-  // success, otherwise it is empty.
+  /// `TextSectionIndex` is specified, only returns the BB address maps
+  /// corresponding to the section with that index. When `PGOAnalyses`is
+  /// specified, the vector is cleared then filled with extra PGO data.
+  /// When PGO information is requested (`PGOAnalyses` is not nullptr),
+  /// `PGOAnalyses` will always be the same length as the return value,
+  /// assuming no error occurs. Upon failure, `PGOAnalyses` will be empty.
```
Comment (rlavaee) [inline llvm/include/llvm/Object/ELFObjectFile.h:118]: will be erased/emptied/cleared.

**Comment by rlavaee** [comment]
LGTM with a minor nit.

File: llvm/include/llvm/Object/ELFObjectFile.h [hunk 2]
```diff
@@ -110,11 +110,12 @@ class ELFObjectFileBase : public ObjectFile {
...
-  // corresponding to the section with that index. When `PGOAnalyses`is
-  // specified, the vector is cleared then filled with extra PGO data.
-  // `PGOAnalyses` will always be the same length as the return value on
-  // success, otherwise it is empty.
+  /// `TextSectionIndex` is specified, only returns the BB address maps
+  /// corresponding to the section with that index. When `PGOAnalyses`is
+  /// specified, the vector is cleared then filled with extra PGO data.
+  /// When PGO information is requested (`PGOAnalyses` is not nullptr),
+  /// `PGOAnalyses` will always be the same length as the return value,
+  /// assuming no error occurs. Upon failure, `PGOAnalyses` will be emptied.
```
Comment (red1bluelost) [inline llvm/include/llvm/Object/ELFObjectFile.h:118]: Could we merge the ``When `PGOAnalyses` is specified, ...`` and ``When PGO information is requested (`PGOAnalyses` is not nullptr),``. Seeing two very similar phrases start of two consecutive sentences threw me off. Maybe just have it on the first sentence or combine the sentences.

**Review by red1bluelost — COMMENTED** [review]
LGTM with one nit about the comment

File: llvm/include/llvm/Object/ELFObjectFile.h (hunk 2 above)
Comment (boomanaiden154) [reply llvm/include/llvm/Object/ELFObjectFile.h:118]: Good point. Reworded it to only keep that phrasing in the first sentence.

**Review by red1bluelost — APPROVED** [review]
LGTM

**Pull request merged**